lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs

```diff
@@ -123,6 +123,19 @@ public void CreateDirectory(Uri directoryUri)
         {
         }
 
+        /// <summary>
+        /// Checks if uri is a directory uri.
+        /// </summary>
+        /// <param name="uri">uri of the directory/file</param>
+        /// <returns>true if uri is for a directory else false</returns>
+        public bool IsDirectory(Uri uri)
+        {
+            var path = uri.AbsolutePath.Trim('/');
```

Comment:
Why does the uri need to trim the '/'?

Reply:
The ListBlobsSegmented does not work is there is a '/' at the beginining of the path. I have changed to AbsolutePath.TrimStart('/')

---
In reply to: [195886086](https://github.com/apache/reef/pull/1473#discussion_r195886086) [](ancestors = 195886086)

Reply:
ok thanks

---
In reply to: [196934429](https://github.com/apache/reef/pull/1473#discussion_r196934429) [](ancestors = 196934429,195886086)